ArchiSteamFarm/NLog/ArchiLogger.cs

```diff
@@ -151,6 +152,20 @@ public sealed class ArchiLogger {
 			Logger.Log(logEventInfo);
 		}
 
+		internal void LogInvite(ulong steamID, EAccountType accountType,  [CallerMemberName] string? previousMethodName = null) {
+			if ((accountType == EAccountType.Invalid) || (steamID == 0)) {
+				throw new ArgumentNullException(nameof(accountType) + " || " + nameof(steamID));
+			}
+
+			string loggedMessage = previousMethodName + "() " + Enum.GetName(typeof(EAccountType), accountType) + " " + Strings.Invite + " " + steamID.ToString();
+
+			LogEventInfo logEventInfo = new LogEventInfo(LogLevel.Trace, Logger.Name, loggedMessage.ToString());
+			logEventInfo.Properties["SteamID"] = steamID;
+			logEventInfo.Properties["AccountType"] = Enum.GetName(typeof(EAccountType), accountType);
```

Comment:
just do `accountType.ToString()`, `GetName()` is very inefficient.

Reply:
Do I even need this line if I add SteamID as a property? Will NLog handle it as object or convert to ulong in this case?

Reply:
If you want numeric type, check what underlying type `EAccountType` is of (probably either `ulong,` `uint` or `int`) and just cast to that. `logEventInfo.Properties["AccountType"] = ((ulong) accountType).ToString()` is a very good approach (and very fast).

Reply:
You should however avoid casting and `ToString()` on things you don't need. NLog probably accepts `object` here, so you can just pass `AccountType` as-is, people can do the parsing to numeric value OR name value themselves. Don't take away that possibility from them for no reason.

Reply:
It does not seems like accepting objects. At least `${event-properties:item=SteamID.AccountType}` does not seem to work at all. And if it's not an object then sting will be more informative for logging purposes. If anyone needs to handle it instead - they should make plugins. Or, as an option, I can make two properties and log both value and name.

Reply:
I'm not saying that you should pass a `SteamID` there, pass `SteamID.AccountType` - enum object, and leave the eventual conversion of that to number or string to the user. This should work.

Reply:
But how user will convert it to a string?

Reply:
See how I do `ToString()` on `Exception` object in default ASF logging:

```xml
    <target xsi:type="ColoredConsole" name="ColoredConsole" layout="${date:format=yyyy-MM-dd HH\:mm\:ss}|${processname}-${processid}|${level:uppercase=true}|${logger}|${message}${onexception:inner= ${exception:format=toString,Data}}" />
```

What stops you from doing the same on enum object to get a friendly string?

Reply:
`${AccountType:format=toString,Data}`

I didn't check if it'll work, but it should.

Reply:
but it's not `${AccountType}`, it's `${event-properties:item=AccountType}`. How to add format there? [Nlog wiki](https://github.com/NLog/NLog/wiki/Event-Context-Layout-Renderer) has nothing about it.

Reply:
Have you tried `${event-properties:item=AccountType:format=toString,Data}`?

Reply:
See here, they're doing something very similar:

![image](https://user-images.githubusercontent.com/1069029/95260976-e561b800-0829-11eb-8ec5-81605a91cb72.png)

Reply:
> format - Format for conversion from object to string. Introduced in NLog 4.1.

It's also documented, just without a lot of examples.

Reply:
`${event-properties:item=AccountType:format=toString,Data}` does nothing.

Reply:
Launch logging of internal NLog log and see why, maybe I screwed up the syntax, it should definitely be possible, if not you can probably open issue on NLog and simply ask if it's possible to render `enum` variable passed in event properties as string, they'll likely be happy to answer.

Reply:
And if they say that it's not possible, then we'll do it ourselves with `AccountTypeText` and `AccountType`, first one doing `ToString()` on enum value, second one with a cast to integer.

Reply:
But I'm PRETTY sure this should be possible, and if not, maybe a valid feature request for another hacktoberfest? 😁

Reply:
It seems that by default it's rendered to text, so I will just leave it as an object. You are both pretty sure that converting it to number should be possible, so I won't waste time on finding HOW to do it.